tests/infrastructure/src/main/java/org/mule/test/infrastructure/process/rules/MuleDeployment.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.infrastructure.process.rules;
+
+import static java.lang.Boolean.parseBoolean;
+import static java.lang.Integer.parseInt;
+import static java.lang.System.getProperty;
+import static org.apache.commons.io.FileUtils.deleteDirectory;
+import static org.apache.commons.io.FileUtils.deleteQuietly;
+import static org.apache.commons.io.FileUtils.moveDirectory;
+import static org.apache.commons.io.FilenameUtils.removeExtension;
+import static org.apache.commons.lang.StringUtils.isEmpty;
+import static org.mule.test.infrastructure.process.AppDeploymentProbe.isDeployed;
+import org.mule.tck.probe.PollingProber;
+import org.mule.test.infrastructure.process.MuleProcessController;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang.StringUtils;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * JUnit rule to deploy a Mule application for testing in a local Mule server. Usage:
+ * <p>
+ * <pre>
+ * public static class MuleDeployment {
+ *  &#064;Rule
+ *  public static MuleDeployment deployment = application(&quot;/path/to/application.zip&quot;).withProperty("-M-Dproperty", "value").timeout(120).deploy();
```

Comment:
What is your suggestion? I consider that a user should know how to use JUnit rules before using it. Besides the most usual use case would be to use a class rule because you will run several tests against the same application.